pandas/tests/indexes/test_datetimelike.py

```diff
@@ -721,6 +721,17 @@ def test_fillna_datetime64(self):
                            dtype=object)
             self.assert_index_equal(idx.fillna('x'), exp)
 
+    def test_contains(self):
+        #GH13572
+        dates = ['2015-01-03', '2015-01-01', '2015-01-04', '2015-01-05', '2015-01-02']
+        monotonic = pd.to_datetime(sorted(dates))
+        non_monotonic = pd.to_datetime(['2015-01-03', '2015-01-01', '2015-01-04', '2015-01-05', '2015-01-02'])
```

Comment:
Hmm, TimedeltaIndices have a funny behavior as well. Checking for NaT-like objects always returns true
```python
>>> None in pd.to_timedelta(range(5), unit='d') + pd.offsets.Hour(1)
True
```
Looks like `pandas.index.TimedeltaEngine.get_loc` doesn't handle NaT at all. It's probably possible to patch it on a higher level, but instinctively I feel that is where it should be handled.
@TomAugspurger / @jreback any thoughts?